src/user/user.spec.js

```diff
@@ -0,0 +1,89 @@
+/* eslint-env jest */
+
+import { omit } from "lodash";
+import { testConnection, xo } from "../util";
+import { createUser, deleteAllUsers, getUser } from "./helpers";
+
+const SUCCESSFUL_TESTS = [
+  [
+    "creates a user without Permission",
+    {
+      email: "wayne1@vates.fr",
+      password: "batman1",
+    },
+  ],
+  [
+    "creates a user with Permission",
+    {
+      email: "wayne2@vates.fr",
+      password: "batman2",
+      permission: "user",
+    },
+  ],
+];
+
+const FAILURE_TESTS = [
+  [
+    "tries to create without email",
+    {
+      password: "batman",
+    },
+  ],
+  [
+    "tries to create without password",
+    {
+      email: "wayne@vates.fr",
+    },
+  ],
+];
+
+const FAILURE_TEST_FOR_EXISTENT_EMAIL = [
+  [
+    "tries to create two users with the same email",
+    {
+      email: "wayne3@vates.fr",
+      password: "batman",
+    },
+  ],
+];
+
+describe("user", () => {
+  afterAll(async () => {
+    await deleteAllUsers(xo);
+  });
+
+  describe(".create()", () => {
+    describe.each(SUCCESSFUL_TESTS)(
+      "create a user successfully",
+      (title, data) => {
+        it(title, async () => {
+          const userId = await createUser(xo, data);
+          expect(userId).toBeString();
+          expect(omit(await getUser(xo, userId), "id")).toMatchSnapshot();
+          await testConnection({
+            credentials: {
+              email: data.email,
+              password: data.password,
+            },
+          });
+        });
+      }
+    );
+
+    describe.each(FAILURE_TESTS)("failed to create a user", (title, data) => {
+      it(title, async () => {
+        await expect(createUser(xo, data)).rejects.toMatchSnapshot();
+      });
+    });
+
+    describe.each(FAILURE_TEST_FOR_EXISTENT_EMAIL)(
```

Comment:
`describe.each` is not needed.

Reply:
I think you can merge this test with the above tests and do a `BeforeAll` for user creation.

Reply:
OK :+1: